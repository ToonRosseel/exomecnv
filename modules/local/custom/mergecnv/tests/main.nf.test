nextflow_process {

    name "Test Process CUSTOM_MERGECNV"
    script "../main.nf"
    process "CUSTOM_MERGECNV"

    test("homo_sapiens - cnv_auto, cnv_chrx") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.cnv_auto, checkIfExists:true),
                    file(params.cnv_chrx, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("homo_sapiens - cnv_auto, cnv_chrx - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'Sample1', chr:"autosomal", sam:["Sample1", "Sample2", "Sample3", "Sample4", "Sample5"], fam:["Fam1", "Fam2", "Fam3", "Fam4", "Fam5"] ],
                    file(params.cnv_auto, checkIfExists:true),
                    file(params.cnv_chrx, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
