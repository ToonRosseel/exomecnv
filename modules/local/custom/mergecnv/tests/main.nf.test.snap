{
    "homo_sapiens - cnv_auto, cnv_chrx": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_CNVs_ExomeDepth.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "1": [
                    "versions.yml:md5,b318789ed32522a7e1c3eb55f4aa5c3e"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_CNVs_ExomeDepth.txt:md5,cd414909ef75fbabb7a3e33b0ae13ea4"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,b318789ed32522a7e1c3eb55f4aa5c3e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T12:15:39.02769988"
    },
    "homo_sapiens - cnv_auto, cnv_chrx - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_CNVs_ExomeDepth.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,4208993fcbca7b639df1108e0a4c7563"
                ],
                "merge": [
                    [
                        {
                            "id": "Sample1",
                            "chr": "autosomal",
                            "sam": [
                                "Sample1",
                                "Sample2",
                                "Sample3",
                                "Sample4",
                                "Sample5"
                            ],
                            "fam": [
                                "Fam1",
                                "Fam2",
                                "Fam3",
                                "Fam4",
                                "Fam5"
                            ]
                        },
                        "Sample1_CNVs_ExomeDepth.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4208993fcbca7b639df1108e0a4c7563"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-26T12:15:46.189890108"
    }
}